Breeding-Insight/bi-api
# [BI-1830] Overwrite entity dataset

# Description
**Story:** [BI-1830](https://breedinginsight.atlassian.net/jira/software/c/projects/BI/boards/1?selectedIssue=BI-1830)

The expected behavior of a user-supplied observation unit dbid superseding experiment, environment, and unit import values has apparently regressed. Fixing this regression will be handled in a separate card, and the parts of `ExperimentProcessor` involved commented out with TODOs. For this feature, observations are updated by filling out the complete import row with the ObsUnitID value blank.

1. Fields `overwrite` and `overwriteReason` added to `ExperimentObservation`
2. validation of existing observations updated
3. a `changeLog` entry is added to additionalInfo when committing updates to existing observations
4. `postBrapiData` was updated to both post new observations and put updates to existing observations



# Dependencies
bi-web `feature/BI-1265`

# Testing
1. import germplasm, observation variables, and a new experiment
2. alter some of the rows of the experiment import file
- change the values of a prior observation
- leave the rest (environment, experiment, unit #, etc.)of the import row the same
3. import the new file
4. view details of the experiment and verify any changes to observation values.


# Checklist:



[BI-1830]: https://breedinginsight.atlassian.net/browse/BI-1830?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
